server/budgettracker.web/Controllers/BudgetApiController.cs

```diff
@@ -21,9 +24,26 @@ public BudgetApiController(IBudgetApi budgetApi)
         }
 
         [HttpPost("create")]
-        public async Task<ApiResponse> CreateBudget(ApiRequest request)
+        public async Task<IActionResult> CreateBudget(ApiRequest request)
         {
-            return await _budgetApi.CreateBudget(request);
+            if(!request.Arguments<CreateBudgetRequestContract>().IsValid())
+            {
+                return BadRequest();
+            }
+
+            try
+            {
+                return Ok(await _budgetApi.CreateBudget(request));    
+            }
+            catch (Exception ex) when (ex.InnerException is AuthenticationException)
+            {
+                if (ex is AuthenticationException) 
+                {
+                    return Forbid();
+                }
+                // We should consider logging the exception here. 
```

Comment:
I agree. Let's add an issue to add logging to the project.

Reply:
However, I don't think we should log this here. I think we shouldn't even catch the exception if it's not an AuthenticationException. If it's something else, let it bubble up to the server to handle (which by default will log it if we set up logging).

Reply:
This should only catch AuthenticationException. Anything else should continue to throw up the stack.